### ISSUE TITLE
'Error running install.py for extension...' during extension installation.

### ISSUE THREAD

**Issue by 925-Studio** [Mikubill/sd-webui-controlnet#240]
I've tried both way to install this extension, from url and available, neither of them worked, keeps showing me the same error:
```Error running install.py for extension /Users/silver/Playground/AI/sdw/extensions/sd-webui-controlnet.
Command: "/Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10" "/Users/silver/Playground/AI/sdw/extensions/sd-webui-controlnet/install.py"
Error code: 1
stdout: Installing sd-webui-controlnet requirement: svglib
...
```

I didn't notice this error at first time. Later I found this extension doesn't work at all then I check the terminal and saw the error.
I'm using mac mini m1 with 16gb ram, ventura 13.2.1. Everything works fine except this extension. 

Can you help me with this? Thanks for you efforts, maybe things are not working out for me, but this is a great extension without question.

**Comment by Mikubill** [comment]
Try manually install svglib using
pip install svglib

Check if any error occurred

**Comment by 925-Studio** [comment]
> Try manually install svglib using pip install svglib
> 
> Check if any error occurred

Wow, you are the man. I just want to update what just happened and your reply is already here. The thing is, after restart the terminal and several times for restart the webui, it's working now. Haven't try `pip install svglib` yet.

**Comment by 925-Studio** [comment]
I'll be more specific here:
1. Quit terminal by CMD+Q and restart a new one
2. `./webui.sh` and same error info appears in terminal
3. Reboot the UI in setting tab and no errors and I can use controlnet
4. I've tried to manually install `svglib` but lots of error showed up, I copied some in below

```
❯ pip3 install svglib
Collecting svglib
  Using cached svglib-1.5.1-py3-none-any.whl
Requirement already satisfied: tinycss2>=0.6.0 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from svglib) (1.2.1)
Requirement already satisfied: reportlab in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from svglib) (3.6.12)
Collecting cssselect2>=0.2.0
  Using cached cssselect2-0.7.0-py3-none-any.whl (15 kB)
Collecting lxml
  Using cached lxml-4.9.2.tar.gz (3.7 MB)
  Preparing metadata (setup.py) ... done
Requirement already satisfied: webencodings in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from cssselect2>=0.2.0->svglib) (0.5.1)
Requirement already satisfied: pillow>=9.0.0 in /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages (from reportlab->svglib) (9.4.0)
Building wheels for collected packages: lxml
  Building wheel for lxml (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [119 lines of output]
      Building lxml version 4.9.2.
      Building without Cython.

......

copying src/lxml/isoschematron/resources/xsl/iso-schematron-xslt1/iso_schematron_message.xsl -> build/lib.macosx-10.9-universal2-cpython-310/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
      copying src/lxml/isoschematron/resources/xsl/iso-schematron-xslt1/readme.txt -> build/lib.macosx-10.9-universal2-cpython-310/lxml/isoschematron/resources/xsl/iso-schematron-xslt1
      running build_ext
      building 'lxml.etree' extension
      creating build/temp.macosx-10.9-universal2-cpython-310
      creating build/temp.macosx-10.9-universal2-cpython-310/src
      creating build/temp.macosx-10.9-universal2-cpython-310/src/lxml
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch arm64 -arch x86_64 -g -I/opt/homebrew/opt/node@16/include -DCYTHON_CLINE_IN_TRACEBACK=0 -Isrc -Isrc/lxml/includes -I/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10 -c src/lxml/etree.c -o build/temp.macosx-10.9-universal2-cpython-310/src/lxml/etree.o -w -flat_namespace
      xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
      Compile failed: command '/usr/bin/clang' failed with exit code 1
      cc -I/usr/include/libxml2 -c /var/folders/1h/s6b1v8xd1gzfpwjydmyh73680000gn/T/xmlXPathInitjevdhujo.c -o var/folders/1h/s6b1v8xd1gzfpwjydmyh73680000gn/T/xmlXPathInitjevdhujo.o
      xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
      *********************************************************************************
      Could not find function xmlCheckVersion in library libxml2. Is libxml2 installed?
      Perhaps try: xcode-select --install
      *********************************************************************************
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> lxml

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

**Comment by 925-Studio** [comment]
Update:
1. I tried to install this extension on my mbp which running big sur and it went smooothly, so I guess maybe it's my command line tool has problem or miss something.
2. According to the error log, I installed `command line developer tools` to give it a shot, and it WORKED! 
3. After installation of `command line developer tools`, I ran the `webui.sh` and it took a pause at `Installing requirements for Web UI`, apparently it was installing the missing piece(s), and couple seconds later, `127.0.0.1:7860` appears and no more errors.

I hope this can be helpful for someone is facing this problem.
